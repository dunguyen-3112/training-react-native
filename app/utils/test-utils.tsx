import { render, RenderOptions } from '@testing-library/react-native';
import React, { ReactElement } from 'react';
import { NavigationContainer } from '@react-navigation/native';

const AllTheProviders = ({ children }: { children: ReactElement }) => {
  return <NavigationContainer>{children}</NavigationContainer>;
};

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react-native';

export { customRender as render };
