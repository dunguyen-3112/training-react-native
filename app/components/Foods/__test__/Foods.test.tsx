import { render, screen, fireEvent } from '@utils/test-utils';
import React from 'react';
import { FoodsProps, FoodsHorizontal } from '../index';

import { MOCK_FOODS } from '@__mock__';

const mockFoodOnPress = jest.fn();

function setup(props: FoodsProps) {
  return render(<FoodsHorizontal {...props} />);
}

describe('Component', () => {
  it('renders correctly', () => {
    const component = setup({ foods: MOCK_FOODS });

    expect(component).toMatchSnapshot();

    MOCK_FOODS.forEach(({ name }) => {
      if (name) expect(screen.getByText(name)).toBeTruthy();
    });

    // const item = MOCK_FOODS[1];
    // if (item.name) {
    //   const select = component.getByText(item.name);
    //   if (select) {
    //     fireEvent.press(select);
    //     expect(mockFoodOnPress).toHaveBeenCalledWith(item.id);
    //   }
    // }
  });
});
