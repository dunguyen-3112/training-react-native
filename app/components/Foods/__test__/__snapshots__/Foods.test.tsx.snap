// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
      },
      Object {
        "alignItems": "center",
      },
    ]
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    getItem={[Function]}
    getItemCount={[Function]}
    justifyContent="space-between"
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    style={
      Object {
        "marginTop": 15,
      }
    }
    viewabilityConfigCallbackPairs={Array []}
    width={326}
  >
    <View />
  </RCTScrollView>
</View>
`;
